File: tests/core/src/main/java/msgsim/commands.java
/*
 * Copyright (c) 2010-2015 Pivotal Software, Inc. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you
 * may not use this file except in compliance with the License. You
 * may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
 * implied. See the License for the specific language governing
 * permissions and limitations under the License. See accompanying
 * LICENSE file.
 */
package msgsim;

public interface commands {

  public static final int BASE_PORT = 15963; // servers use ports 15963...

  // these are messaging commands - they are read and executed by "reader" threads
  public static enum MessageProtocolCommand {
    INVALID((byte) -1),
    INIT((byte) 0),
    USE_THREAD_OWNED((byte) 1),
    USE_SHARED((byte) 2),
    HAND_OFF_LISTENERS((byte) 3),
    USE_THREAD_OWNED_UNORDERED((byte) 4),
    USE_POOLED_CONNECTIONS((byte) 5),
    NORMAL_MESSAGE((byte) 10),
    SET_PAYLOAD_SIZE((byte) 11),
    ACK_MESSAGE((byte) 12),
    PRINT_QUEUE_SIZES((byte) 13),
    EXIT((byte) 127);

    private final byte byteValue;

    MessageProtocolCommand(final byte byteValue) {
      this.byteValue = byteValue;
    }

    public static MessageProtocolCommand valueOf(final byte byteValue) {
      for (final MessageProtocolCommand command : values()) {
        if (command.toByte() == byteValue) {
          return command;
        }
      }

      return INVALID;
    }

    public boolean isInvalid() {
      return this.equals(INVALID);
    }

    public byte toByte() {
      return byteValue;
    }
  }

  // These are scenarios that go with the "scenario" command line argument.  The default is SERVER.
  public static enum MessagingScenario {
    UNKNOWN(-1, "Unknown"),
    SERVER(0, "Server"),
    SHARED_CONNECTIONS(1, "conserve-sockets=true"),
    SHARED_CONNECTIONS_WITH_DOMINO(2, "conserve-sockets=true, readers get owned ack sockets (conserve-sockets=false"),
    THREAD_OWNED_CONNECTIONS(3, "conserve-sockets=false"),
    THREAD_OWNED_CONNECTIONS_WITH_DOMINO(4, "conserve-sockets=false, readers get conserve-sockets=false"),
    POOLED_CONNECTIONS(5, "queued connection pool"),
    POOLED_CONNECTIONS_WITH_DOMINO(6, "queued connection pool with domino messages"),
    HANDOFF_LISTENERS(7, "conserve-sockets=true, listener-handoff"),
    HANDOFF_LISTENERS_WITH_DOMINO(8, "conserve-sockets=true, readers get owned ack sockets (conserve-sockets=false"),
    EXIT(999, "Shutdown");

    private final int id;
    private final String description;

    MessagingScenario(final int id, final String description) {
      this.id = id;
      this.description = description;
    }

    public static MessagingScenario getDefault() {
      return SERVER;
    }

    public static MessagingScenario valueOf(final int id) {
      for (final MessagingScenario scenario : values()) {
        if (scenario.getId() == id) {
          return scenario;
        }
      }

      return UNKNOWN;
    }

    public String getDescription() {
      return description;
    }

    public int getId() {
      return id;
    }

    public boolean isExit() {
      return this.equals(EXIT);
    }

    public boolean isServer() {
      return this.equals(SERVER);
    }

    @Override
    public String toString() {
      return getDescription();
    }
  }

}

File: gemfirexd/client/test/csharp/ado-net-gemfirexd-client/Utilities/Commands.cs
/*
 * Copyright (c) 2010-2015 Pivotal Software, Inc. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you
 * may not use this file except in compliance with the License. You
 * may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
 * implied. See the License for the specific language governing
 * permissions and limitations under the License. See accompanying
 * LICENSE file.
 */
 
using System;
using System.Collections.Generic;
using System.Data;
using System.IO;
using System.Linq;
using System.Text;
using System.Windows.Input;
using System.Xml;

namespace ado_net_gemfirexd_client.Utilities
{
    public static class Commands
    {
        public static readonly RoutedUICommand NewSalesOrder;
        public static readonly RoutedUICommand DeleteSalesOrder;
        public static readonly RoutedUICommand UpdateSalesOrder;

        public static readonly RoutedUICommand NewCustomer;
        public static readonly RoutedUICommand DeleteCustomer;
        public static readonly RoutedUICommand UpdateCustomer;

        public static readonly RoutedUICommand NewProduct;
        public static readonly RoutedUICommand DeleteProduct;
        public static readonly RoutedUICommand UpdateProduct;

        static Commands()
        {
            NewSalesOrder = new RoutedUICommand("Create a new Sales Order", "NewSalesOrder", typeof(Commands));
            DeleteSalesOrder = new RoutedUICommand("Delete current Sales Order", "DeleteSalesOrder", typeof(Commands));
            UpdateSalesOrder = new RoutedUICommand("Update current Sales Order", "UpdateSalesOrder", typeof(Commands));

            NewCustomer = new RoutedUICommand("Create a new customer", "NewCustomer", typeof(Commands));
            DeleteCustomer = new RoutedUICommand("Delete current customer", "DeleteCustomer", typeof(Commands));
            UpdateCustomer = new RoutedUICommand("Update current customer", "UpdateCustomer", typeof(Commands));

            NewProduct = new RoutedUICommand("Create a new product", "NewProduct", typeof(Commands));
            DeleteProduct = new RoutedUICommand("Delete current product", "DeleteProduct", typeof(Commands));
            UpdateProduct = new RoutedUICommand("Update current product", "UpdateProduct", typeof(Commands));
        }


        public static string ToString(DataSet dsData)
        {
            var objStream = new MemoryStream();
            dsData.WriteXml(objStream);

            var objXmlWriter = new XmlTextWriter(objStream, Encoding.UTF8);
            objStream = (MemoryStream)objXmlWriter.BaseStream;

            var objEncoding = new UTF8Encoding();

            return objEncoding.GetString(objStream.ToArray());
        }
    }
}
